source/adios2/core/ADIOS.cpp

```diff
@@ -79,6 +79,18 @@ IO &ADIOS::DeclareIO(const std::string ioName)
     return ioPair.first->second;
 }
 
+IO &ADIOS::GetIO(const std::string name)
+{
+    auto itIO = m_IOs.find(name);
+    if (itIO == m_IOs.end())
```

Comment:
There won't be an exception since there's no code inside libstdc++ that's testing for an error case and throwing one, which is different from the new operator which checks for success or failure of memory allocation and can throw a bad_alloc exception when detected.  I don't think that we can allow an access violation when we know it's going to occur.  In this case, our software stack is in a known state and can easily proceed and recover.